client/client.go

```diff
@@ -462,6 +462,46 @@ func (r *NotaryRepository) GetTargetByName(name string, roles ...string) (*Targe
 
 }
 
+// CheckTargetByName returns true if all passed-in roles have signed the specified target with the same hash.
+// If no roles are passed, it uses the targets role.  Note that child roles are considered if
+// we can't find the target in the specified role.
+func (r *NotaryRepository) CheckTargetByName(name string, roles ...string) (bool, error) {
+	if err := r.Update(false); err != nil {
+		return false, err
+	}
+
+	if len(roles) == 0 {
```

Comment:
No problem, I agree that it's good to discuss and think this through -- I'm totally open to changing this function if we prefer different behavior if it improves the API :)